<MonoAOTCompiler/> MSBuild task does not work when used as <PackageReference/>

### Description

If I have a project such as:

```xml
<Project Sdk="Microsoft.Build.NoTargets/2.0.17">
  <ItemGroup>
    <PackageReference Include="Microsoft.NET.Runtime.MonoAOTCompiler.Task" Version="6.0.0-preview.1.21072.5" IncludeAssets="build" />
  </ItemGroup>
  <UsingTask TaskName="MonoAOTCompiler" AssemblyFile="$(MonoAOTCompilerTaskAssemblyPath)" />
  <Target Name="Build">
    <!-- NOTE: just trying to get to the point the task is called -->
    <MonoAOTCompiler />
  </Target>
</Project>
```
(You could also use `Microsoft.NET.Sdk` if you prefer)

With the `nuget.config`:
```xml
<?xml version="1.0" encoding="utf-8"?>
<configuration>
  <packageSources>
    <add key="dotnet6" value="https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json" />
  </packageSources>
</configuration>
```

Building fails with:

```
error MSB4022: The result "" of evaluating the value "$(MonoAOTCompilerTaskAssemblyPath)" of the "AssemblyFile" attribute in element <UsingTask> is not valid.
```

Log: [msbuild.zip](https://github.com/dotnet/runtime/files/5894423/msbuild.zip)

### Configuration

On Windows 10:
```
> dotnet --version
6.0.100-alpha.1.21064.27
```

### Regression?

No

### Other information

I think the idea of using `$(MonoAOTCompilerTaskAssemblyPath)` with `<PackageReference/>` and `<UsingTask/>` might be flawed. I think `<UsingTask/>` will be evaluated before `$(MonoAOTCompilerTaskAssemblyPath)` is set.

Let me know if I missed something, it seems like I'm using it as the doc suggests:

https://github.com/dotnet/runtime/blob/5fc66480fe47ff8f4b2620901e2d33177af2dd4c/src/mono/netcore/nuget/Microsoft.NET.Runtime.MonoAOTCompiler.Task/README.md

We got this to work in the upcoming Android workload by installing this package as a [workload pack](https://github.com/xamarin/xamarin-android/pull/5539). So we aren't blocked on this, thanks!

--- a/src/mono/nuget/Microsoft.NET.Runtime.MonoAOTCompiler.Task/README.md
+++ b/src/mono/nuget/Microsoft.NET.Runtime.MonoAOTCompiler.Task/README.md
@@ -40,7 +40,7 @@ Use `UsingTask` directly
 </ItemGroup>
 
 <UsingTask TaskName="MonoAOTCompiler"
-           AssemblyFile="$(PkgMicrosoft_NET_Runtime_MonoAOTCompiler_Task)/tools/$(TargetFramework)/MonoAOTCompiler.dll" />
+           AssemblyFile="$(PkgMicrosoft_NET_Runtime_MonoAOTCompiler_Task)/tools/<target-framework>/MonoAOTCompiler.dll" />
 
 <!-- Use the MonoAOTCompiler task in a target -->
 <Target>

--- a/src/mono/nuget/Microsoft.NET.Runtime.MonoAOTCompiler.Task/README.md
+++ b/src/mono/nuget/Microsoft.NET.Runtime.MonoAOTCompiler.Task/README.md
@@ -29,3 +29,27 @@ The `MonoAOTCompiler` MSBuild task is also useful outside the context of `dotnet
     </MonoAOTCompiler>
 </Target>
 ```
+
+------------
+Use `UsingTask` directly 
+## In the project file
+```
+<!-- Import the NuGet package into the project -->
+<ItemGroup>
+    <PackageReference Include="Microsoft.NET.Runtime.MonoAOTCompiler.Task" Version="<desired-dotnet-6-sdk-version>" GeneratePathProperty="true" />
+</ItemGroup>
+
+<UsingTask TaskName="MonoAOTCompiler"
+           AssemblyFile="$(PkgMicrosoft_NET_Runtime_MonoAOTCompiler_Task)/tools/$(TargetFramework)/MonoAOTCompiler.dll" />
+
+<!-- Use the MonoAOTCompiler task in a target -->
+<Target>
+    <MonoAOTCompiler 
+        CompilerBinaryPath="$(CompilerBinaryPath)"
+        Assemblies="@(Assemblies)"
+        <!-- Other parameters -->
+        >
+        <Output TaskParameter="CompiledAssemblies" ItemName="CompiledAssemblies" />
+    </MonoAOTCompiler>
+</Target>
+```

--- a/src/mono/nuget/Microsoft.NET.Runtime.MonoAOTCompiler.Task/README.md
+++ b/src/mono/nuget/Microsoft.NET.Runtime.MonoAOTCompiler.Task/README.md
@@ -31,7 +31,7 @@ The `MonoAOTCompiler` MSBuild task is also useful outside the context of `dotnet
 ```
 
 ------------
-Use `UsingTask` directly 
+Use `UsingTask` directly
 ## In the project file
 ```
 <!-- Import the NuGet package into the project -->

